src/software/ai/passing/cost_function.cpp

```diff
@@ -8,67 +8,70 @@
 
-    double friendly_pass_rating = ratePassFriendlyCapability(
-        world.friendlyTeam(), pass, passer_robot_id, passing_config);
+    double friendly_pass_rating =
+        ratePassFriendlyCapability(world.friendlyTeam(), pass, passing_config);
 
     double enemy_pass_rating = ratePassEnemyRisk(world.enemyTeam(), pass, passing_config);
 
     double shoot_pass_rating =
         ratePassShootScore(world.field(), world.enemyTeam(), pass, passing_config);
 
-    // Rate all passes outside our target region as 0 if we have one
-    double in_region_quality = 1;
-    if (target_region)
-    {
-        in_region_quality = rectangleSigmoid(*target_region, pass.receiverPoint(), 0.1);
-    }
-
-    // Place strict limits on pass start time
-    double min_pass_time_offset =
-        passing_config->getMinTimeOffsetForPassSeconds()->value();
-    double max_pass_time_offset =
-        passing_config->getMaxTimeOffsetForPassSeconds()->value();
-    double pass_time_offset_quality =
-        sigmoid(pass.startTime().toSeconds(),
-                min_pass_time_offset + world.getMostRecentTimestamp().toSeconds(), 0.5) *
-        (1 - sigmoid(pass.startTime().toSeconds(),
-                     max_pass_time_offset + world.ball().timestamp().toSeconds(), 0.5));
+    double in_region_quality = rectangleSigmoid(zone, pass.receiverPoint(), 0.4);
 
     // Place strict limits on the ball speed
     double min_pass_speed     = passing_config->getMinPassSpeedMPerS()->value();
     double max_pass_speed     = passing_config->getMaxPassSpeedMPerS()->value();
     double pass_speed_quality = sigmoid(pass.speed(), min_pass_speed, 0.2) *
                                 (1 - sigmoid(pass.speed(), max_pass_speed, 0.2));
 
-    // We ignore shoot pass quality if we're rating for a "RECEIVE_AND_DRIBBLE" pass
-    double pass_quality = 0;
-    switch (pass_type)
-    {
-        case PassType::RECEIVE_AND_DRIBBLE:
-            pass_quality = static_pass_quality * friendly_pass_rating *
-                           enemy_pass_rating * in_region_quality *
-                           pass_time_offset_quality * pass_speed_quality;
-            break;
-        case PassType::ONE_TOUCH_SHOT:
-            pass_quality = static_pass_quality * friendly_pass_rating *
-                           enemy_pass_rating * shoot_pass_rating * in_region_quality *
-                           pass_time_offset_quality * pass_speed_quality;
-            break;
-        default:
-            throw std::invalid_argument("Unhandled pass type given to `ratePass`");
-    }
+    return static_pass_quality * friendly_pass_rating * enemy_pass_rating *
+           shoot_pass_rating * pass_speed_quality * in_region_quality;
+}
 
-    return pass_quality;
+double rateZone(const Field& field, const Team& enemy_team, const Rectangle& zone,
+                const Point& ball_position,
+                std::shared_ptr<const PassingConfig> passing_config)
```

Comment:
I just wanted to implement something quickly that de-risks automatic zone assignment and see if its viable with this function. Its by no means the only function we will need for _any_ pitch division

Since this is not the finalized implementation/place for it, I am hesitant to build out tests for it right now. I added a TODO in the function and I am going to deffer the work to #2021 to improve this/move it elsewhere and finalize/test it properly.